plugins/neo4j-dump-widget/core/CoreExtension.js

```diff
@@ -0,0 +1,61 @@
+export class CoreExtension {
+  #core
+
+  constructor(core) {
+    this.#core = core
+  }
+
+  useExtension() {
+    const neo4jCore = this
+    this.#core.use(
+      class VueCoreNeo4jExtension {
+        static install(Vue) {
+          Vue.prototype.$neo4jCore = neo4jCore
+        }
+      }
+    )
+  }
+
+  static async handleAppError(response) {
+    if (!response.ok) {
+      var error = null;
+      try {
+        const { title, detail, trace } = await response.json()
+        error = this.formatAppError(title, detail, trace)
+      } catch (e) {
+        if (e instanceof TypeError) {
+          error = await response.text()
+        } else {
+          throw e;
+        }
+      }
+      throw new Error(error)
+    }
+    return response
+  }
+
+  static formatAppError(title, detail, trace) {
+    const formatted = `${title}
+Detail: ${detail}
+`
+    return trace ? `${formatted}\n${trace}` : formatted
+  }
+
+  async request(url, config = {}) {
+    const ApiClass = this.#core.api.constructor
+    const fullUrl = ApiClass.getFullUrl(url)
+    try {
+      const r = await fetch(fullUrl, {
```

Comment:
You could use the `api` methods directly:

https://github.com/ICIJ/datashare-client/blob/master/src/api/index.js#L285-L298

For instance:

```js
this.#core.api.sendAction(url, config)
// Or with axios directly
this.#core.api.axios.request(url, config)
```

Reply:
The problem here is that axios won't let me stream the bytes directly even if I specify `responseType: 'stream'`.
What it does instead is creating a string out of bytes streamed by the backend and then return them, which might get the memory blow up in practice.

Reply:
Make sense, thanks!